### PR TITLE
Less hacky fix for /%2F bug

### LINKED ISSUE #804: /api/topology/applications-by-name/%2Fhome%2Fweave%2Fweave gets redirected to /api/topology/applications-by-name/home/weave/weave
A bug in go / gorilla: https://github.com/gorilla/mux/issues/142

I can probably do a workaround.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #804

### CODE REVIEW

File: app/router.go [hunk 1]
```diff
@@ -68,11 +68,28 @@ func RegisterTopologyRoutes(c Reporter, router *mux.Router) {
...
+// means the request has been handled.
+func HandleNode(c Reporter, w http.ResponseWriter, r *http.Request) bool {
+	matcher := URLMatcher("/api/topology/{topology}/{id}")
+	rm := mux.RouteMatch{}
+	if !matcher(r, &rm) {
+		return false
+	}
+	topologyID := rm.Vars["topology"]
+	nodeID := rm.Vars["id"]
+	if nodeID == "ws" {
```
Comment (paulbellamy) [inline app/router.go]: Won't we still have the same bug for websockets, then?

File: prog/app.go [hunk 2]
```diff
@@ -18,14 +18,21 @@ import (
...
 	router.Methods("GET").PathPrefix("/").Handler(http.FileServer(FS(false)))
-	return router
+
+	// need to handle some routes in a completely custom way, because
+	// https://github.com/weaveworks/scope/issues/804
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !app.HandleNode(c, w, r) {
+			router.ServeHTTP(w, r)
+		}
+	})
```
Comment (paulbellamy) [inline prog/app.go]: Don't like that this is in 2 places. Maybe extract it somewhere into a function with the explanation there. Could just inline it into `app.HandleNode`, maybe.

**Comment by paulbellamy** [comment]
ugly, but can't think of a better way to do it.

**Comment by tomwilkie** [comment]
I think this is beyond even my ugliness threshold.  @peterbourgon have any suggestions?  Otherwise I'll take another stab as bashing it into shape.

**Comment by tomwilkie** [comment]
> Won't we still have the same bug for websockets, then?

None of the topology view name start with (or contain) a slash, so we won't have this bug.

> Don't like that this is in 2 places.

Have updated and rejigged this a little so its less ugly.  Also added a test.  @paulbellamy PTAL.

**Pull request merged**